pkg/cli/admin/update/update.go

```diff
@@ -52,64 +52,77 @@ func NewOptions(streams genericclioptions.IOStreams) *Options {
 			kcmdutil.CheckErr(o.Run())
 		},
+		Aliases: []string{"upgrade"},
 	}
 	flags := cmd.Flags()
-	flags.StringVar(&o.To, "to", o.To, "Specify the version to upgrade to. The version must be on the list of available updates.")
-	flags.StringVar(&o.ToImage, "to-image", o.ToImage, "Provide a release image to upgrade to.")
+	flags.StringVar(&o.To, "to", o.To, "Specify the version to update to. The version must be on the list of available updates.")
+	flags.StringVar(&o.ToImage, "to-image", o.ToImage, "Provide a release image to update to.")
 	flags.BoolVar(&o.ToLatestAvailable, "to-latest", o.ToLatestAvailable, "Use the next available version.")
-	flags.BoolVar(&o.ToMultiArch, "to-multi-arch", o.ToMultiArch, "Upgrade current version to multi architecture.")
-	flags.BoolVar(&o.Clear, "clear", o.Clear, "If an upgrade has been requested but not yet downloaded, cancel the update. This has no effect once the update has started.")
-	flags.BoolVar(&o.Force, "force", o.Force, "Upgrade regardless of cluster-side guard failures, such as release verification or upgradeable conditions.")
-	flags.BoolVar(&o.AllowExplicitUpgrade, "allow-explicit-upgrade", o.AllowExplicitUpgrade, "Upgrade even if the upgrade target is not listed in the available versions list.")
-	flags.BoolVar(&o.AllowUpgradeWithWarnings, "allow-upgrade-with-warnings", o.AllowUpgradeWithWarnings, "Upgrade regardless of client-side guard failures, such as upgrades in progress or failing clusters.")
+	flags.BoolVar(&o.ToMultiArch, "to-multi-arch", o.ToMultiArch, "Update current version to multi architecture.")
+	flags.BoolVar(&o.Clear, "clear", o.Clear, "If an update has been requested but not yet downloaded, cancel the update. This has no effect once the update has started.")
+	flags.BoolVar(&o.Force, "force", o.Force, "Update regardless of cluster-side guard failures, such as release verification or updateable conditions.")
+	flags.BoolVar(&o.AllowExplicitUpdate, "allow-explicit-update", o.AllowExplicitUpdate, "Update even if the update target is not listed in the available versions list.")
+	flags.BoolVar(&o.AllowUpdateWithWarnings, "allow-update-with-warnings", o.AllowUpdateWithWarnings, "Update regardless of client-side guard failures, such as updates in progress or failing clusters.")
 	flags.BoolVar(&o.IncludeNotRecommended, "include-not-recommended", o.IncludeNotRecommended, "Display additional updates which are not recommended based on your cluster configuration.")
-	flags.BoolVar(&o.AllowNotRecommended, "allow-not-recommended", o.AllowNotRecommended, "Allows upgrade to a version when it is supported but not recommended for updates.")
+	flags.BoolVar(&o.AllowNotRecommended, "allow-not-recommended", o.AllowNotRecommended, "Allows update to a version when it is supported but not recommended for updates.")
+
+	flags.BoolVar(&o.AllowExplicitUpdate, "allow-explicit-upgrade", o.AllowExplicitUpdate, "Update even if the update target is not listed in the available versions list.")
```

Comment:
so both the new "update" a well as the old "upgrade" flags variants are expected to work identically, while the later just won't be mentioned anymore?

Reply:
flags.MarkDeprecated is just writing to stderr, or there's other differences expected?

Reply:
also what about autocomplete? will it be able to pick the former and not the latter?

Reply:
[`MarkDeprecated`][1] sets up stderr logging and also hides the flag from help text.

[1]: https://pkg.go.dev/github.com/spf13/pflag#FlagSet.MarkDeprecated